torchaudio/transforms.py

```diff
@@ -70,6 +71,68 @@ def forward(self, waveform):
                              self.win_length, self.power, self.normalized)
 
 
+class GriffinLim(torch.nn.Module):
+    r"""Compute waveform from a linear scale magnitude spectrogram using the Griffin-Lim transformation.
+        Implementation ported from `librosa`.
+
+    .. [1] McFee, Brian, Colin Raffel, Dawen Liang, Daniel PW Ellis, Matt McVicar, Eric Battenberg, and Oriol Nieto.
+        "librosa: Audio and music signal analysis in python."
+        In Proceedings of the 14th python in science conference, pp. 18-25. 2015.
+
+    .. [2] Perraudin, N., Balazs, P., & Søndergaard, P. L.
+        "A fast Griffin-Lim algorithm,"
+        IEEE Workshop on Applications of Signal Processing to Audio and Acoustics (pp. 1-4),
+        Oct. 2013.
+
+    .. [3] D. W. Griffin and J. S. Lim,
+        "Signal estimation from modified short-time Fourier transform,"
+        IEEE Trans. ASSP, vol.32, no.2, pp.236–243, Apr. 1984.
+
+    Args:
+        n_fft (int, optional): Size of FFT, creates ``n_fft // 2 + 1`` bins
+        n_iter (int, optional): Number of iteration for phase recovery process.
+        win_length (int): Window size. (Default: ``n_fft``)
+        hop_length (int, optional): Length of hop between STFT windows. (
+            Default: ``win_length // 2``)
+        window_fn (Callable[[...], torch.Tensor]): A function to create a window tensor
+            that is applied/multiplied to each frame/window. (Default: ``torch.hann_window``)
+        power (int): Exponent for the magnitude spectrogram,
+            (must be > 0) e.g., 1 for energy, 2 for power, etc. (Default: ``2``)
+        normalized (bool): Whether to normalize by magnitude after stft. (Default: ``False``)
+        wkwargs (Dict[..., ...]): Arguments for window function. (Default: ``None``)
+        momentum (float): The momentum parameter for fast Griffin-Lim.
+            Setting this to 0 recovers the original Griffin-Lim method.
+            Values near 1 can lead to faster convergence, but above 1 may not converge. (Default: 0.99)
+        length (int, optional): Array length of the expected output. (Default: ``None``)
+        rand_init(bool): 
+    """
+    __constants__ = ['n_fft', 'n_iter', 'win_length', 'hop_length', 'power', 'normalized',
+                     'length', 'momentum']
+
+    def __init__(self, n_fft=400, n_iter=32, hop_length=None, win_length=None,
+                 window_fn=torch.hann_window, wkwargs=None, normalized=False,
+                 power=2, length=None, momentum=0.99, rand_init=True):
+        super(GriffinLim, self).__init__()
+
+        assert momentum < 1, 'momentum=%s > 1 can be unstable' % momentum
```

Comment:
How about momentum = 1?

nit: the message sounds more like a warning, then an error. Is the algorithm compatible with > 1?

Reply:
I agree, this is more of a warning. I had the assertions set because Perraudin and Sondergaard states that momentum = 1 is the limit of the stability of the algorithm with some results indicating that it fails to converge with momentum > 1 (see figure 3 of P&S). It'll still run and return some (although not ideal) results so if there is some warning mechanism torchaudio is using that would be more appropriate. (Speaking of which, is there any?)

Reply:
Thanks for the clarification! Let's keep the assertion then. 

(For warnings, we often use `warnings.warn`.)